project/plugins/iam.py

```diff
@@ -313,15 +313,63 @@ def store_key_parameter_store(configMap, username,  **key_args):
     if values.DryRun is True:
         logging.info('Dry run: store_key_parameter_store')
     else:
-        client.put_parameter(
-            Name='LOCK.'+username.upper(),
+        if key_args.get('param_name') is not None:
+            parameter_name = key_args.get('param_name')
```

Comment:
Minor thing - but you could avoid this if statement by using a default value in .get, example: key_args.get('param_name', 'LOCK.'+username.upper())

You might even be able to use f'LOCK.{username.upper()} 🙂